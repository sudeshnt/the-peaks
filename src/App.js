import React, { useState } from 'react';
import {
  BrowserRouter,
} from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { ThemeSwitcher } from 'App.styled';
import AppContext from 'AppContext';
import images from 'assets/images';
import Header from 'components/header/Header';
import { SortOrders, StorageKeys } from 'config/shared';
import Router from 'Router';
import GlobalStyles from 'styles/globalStyles';
import { Themes, useTheme } from 'styles/theme/useTheme';

function App() {
  const { theme, themeName, changeTheme } = useTheme();

  const [sortOrder, setSortOrder] = useState(() => {
    const order = localStorage.getItem(StorageKeys.SORT_ORDER);
    // eslint-disable-next-line no-unused-vars
    const sortOrderValid = Object.entries(SortOrders).some(([id, { key }]) => key === order);
    if (sortOrderValid) {
      return order;
    }
    localStorage.setItem(StorageKeys.SORT_ORDER, SortOrders.NEWEST_FIRST.key);
    return SortOrders.NEWEST_FIRST.key;
  });

  const onToggleTheme = () => {
    if (themeName === Themes.LIGHT) {
      changeTheme(Themes.DARK);
    } else if (themeName === Themes.DARK) {
      changeTheme(Themes.LIGHT);
    }
  };

  return (
    <div className="app">
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <BrowserRouter>
          <Header />
          <AppContext.Provider value={{
            sortOrder,
            setSortOrder,
          }}
          >
            <div className="content">
              <Router />
            </div>
          </AppContext.Provider>
        </BrowserRouter>
        <ThemeSwitcher onClick={onToggleTheme}>
          <img src={images.themeSwitch} alt="" />
        </ThemeSwitcher>
      </ThemeProvider>
    </div>
  );
}

export default App;
