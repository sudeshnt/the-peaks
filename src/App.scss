@import "./styles/Base.scss";

h1, h2 {
  font-weight: bold;
}

select {
  border-radius: 0px;
  border: 0;
  border-bottom: 1px solid black;
  outline: none;
  padding: 0 10px;
  width: 180px;
}

.btn {
  display: flex;
  align-items: center;
  position: relative;
  border: none;
  border-radius: 4px;
  height: 35px;
  padding: 5px 12px;
  background: #09357B;
  color: white;
  cursor: pointer;
  text-transform: uppercase;
  font-size: 13px;

  span {
    margin-left: 10px;
  }
}
