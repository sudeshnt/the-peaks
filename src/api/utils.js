import queryString from 'query-string';

export const TopStoriesCount = 8;
export const ArticlesPerSection = 6;

export const NewsSections = Object.freeze({
  NEWS: 'news',
  SPORT: 'sport',
  CULTURE: 'culture',
  LIFE_STYLE: 'lifeandstyle',
});

export const ShowFields = Object.freeze({
  HEADLINE: 'headline',
  THUMBNAIL: 'thumbnail',
  BODY: 'body',
  TRAIL_TEXT: 'trailText',
});

export const commonFields = [ShowFields.HEADLINE, ShowFields.THUMBNAIL];

export const getQueryString = (queryObj = {}) => {
  const queryObject = {
    ...queryObj,
  };
  return queryString.stringify(queryObject);
};

export const getShowFields = (additionalFields = []) => commonFields.concat(additionalFields).join(',');
