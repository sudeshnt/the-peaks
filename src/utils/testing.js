import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

export const renderWithStore = (
  ui,
  initialState = {
    news: {
      sectionNews: [],
    },
    search: {},
    bookmark: {},
  },
) => {
  const mockStore = getMockStore(initialState);
  return {
    ...render(
      <Provider store={mockStore}>
        {ui}
      </Provider>,
    ),
  };
};

const getMockStore = (initialState) => {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  return mockStore(initialState);
};
