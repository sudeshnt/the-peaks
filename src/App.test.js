import { renderWithStore } from './utils/testing';
import App from 'App';

test('renders learn react link', () => {
  const { container } = renderWithStore(<App />);
  expect(container).not.toBeNull();
  expect(container.firstChild).toMatchSnapshot();
});
