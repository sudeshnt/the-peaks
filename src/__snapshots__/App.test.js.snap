// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  class="app"
>
  <header
    class="sc-gtsrHT sc-dlnjwi kUIPJQ gSmzIp app-header"
  >
    <div
      class="content"
    >
      <div
        class="header-logo"
      >
        <img
          class="sc-bdnxRM enAzuc"
          src="logo.png"
        />
      </div>
      <div
        class="search-box"
      >
        <div
          class="sc-jSFjdj kzmeaT"
        >
          <div
            class="input-container"
          >
            <input
              placeholder="Search all news"
              value=""
            />
          </div>
          <button
            class="search-button"
            type="button"
          >
            <img
              alt=""
              src="search-icon.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="content"
  >
    <div
      class="page-content"
    >
      <header
        class="sc-gtsrHT sc-hKFxyN kUIPJQ hWqlPC app-sub-header"
      >
        <div
          class="content"
        >
          <h1>
            Top Stories
          </h1>
          <div
            class="sc-fnVZcZ bhYitd right-container"
          >
            <button
              class="sc-fujyAs gpavy btn"
            >
              <img
                alt=""
                src="bookmarkon-icon.svg"
              />
              <span>
                view bookmarks
              </span>
            </button>
            <select
              class="order-picker"
            >
              <option
                value="newest"
              >
                Newest First
              </option>
              <option
                value="oldest"
              >
                Oldest First
              </option>
            </select>
          </div>
        </div>
      </header>
      <section>
        <div
          class="sc-fFSPTT kjoPor"
          loading="0"
        >
          <div
            class="column left"
          >
            <div
              class="sc-iqAclL fqrKRI article"
              type="WITH_TITLE_THUMBNAIL_AND_DESCRIPTION"
            >
              <div
                class="sc-crzoAE gyEmEE"
              >
                <img
                  alt=""
                  src="logo.png"
                />
              </div>
              <div
                class="sc-dIsUp kRYAA-d"
              >
                <div
                  class="sc-bqGGPW fEjmCQ"
                />
                <div
                  class="sc-ksluID koOAYW"
                >
                  <div />
                </div>
              </div>
            </div>
          </div>
          <div
            class="column right"
          >
            <div
              class="top"
            >
              <div
                class="column"
              >
                <div
                  class="sc-iqAclL fqrKRI article"
                  type="WITH_TITLE_THUMBNAIL_AND_DESCRIPTION"
                >
                  <div
                    class="sc-crzoAE gyEmEE"
                  >
                    <img
                      alt=""
                      src="logo.png"
                    />
                  </div>
                  <div
                    class="sc-dIsUp kRYAA-d"
                  >
                    <div
                      class="sc-bqGGPW fEjmCQ"
                    />
                    <div
                      class="sc-ksluID koOAYW"
                    >
                      <div />
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="column"
              >
                <div
                  class="sc-iqAclL dsyUuF article"
                  type="WITH_TITLE_THUMBNAIL_AND_DESCRIPTION"
                >
                  <div
                    class="sc-crzoAE gyEmEE"
                  >
                    <img
                      alt=""
                      src="logo.png"
                    />
                  </div>
                  <div
                    class="sc-dIsUp kRYAA-d"
                  >
                    <div
                      class="sc-bqGGPW fEjmCQ"
                    />
                    <div
                      class="sc-ksluID koOAYW"
                    >
                      <div />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="bottom"
            >
              <div
                class="column"
              >
                <div
                  class="sc-iqAclL fJJufk article"
                  type="WITH_TITLE"
                >
                  <div
                    class="sc-dIsUp kRYAA-d"
                  >
                    <div
                      class="sc-bqGGPW fEjmCQ"
                    />
                  </div>
                </div>
              </div>
              <div
                class="column"
              >
                <div
                  class="sc-iqAclL hfnpVg article"
                  type="WITH_TITLE"
                >
                  <div
                    class="sc-dIsUp kRYAA-d"
                  >
                    <div
                      class="sc-bqGGPW fEjmCQ"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-bkbkJK jBjwII"
          loading="0"
        />
      </section>
    </div>
  </div>
  <button
    class="sc-eCApnc fOzRso"
  >
    <img
      alt=""
      src="close-icon.svg"
    />
  </button>
</div>
`;
